src/Workspaces/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptAsynchronousOperationListenerProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api;
+
+[Shared]
+[Export(typeof(VSTypeScriptAsynchronousOperationListenerProvider))]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class VSTypeScriptAsynchronousOperationListenerProvider(
+    IAsynchronousOperationListenerProvider provider)
+{
+    public VSTypeScriptAsynchronousOperationListener GetListener(string featureName)
+        => new(provider.GetListener(featureName));
```

Comment:
Unlike the underlying provider, this means multiple calls for the same featureName will return different instances. Since it's stateless though I presume that's fine...

Reply:
Yeah, should be fine. Usually this is called in a ctor and then the listener is used in the type.